src/parquet/column/column-writer-test.cc

```diff
@@ -32,59 +35,93 @@ using schema::PrimitiveNode;
 
 namespace test {
 
+template <typename TestType>
 class TestPrimitiveWriter : public ::testing::Test {
  public:
+  typedef typename TestType::c_type T;
+
   void SetUpSchemaRequired() {
-    node_ = PrimitiveNode::Make("int64", Repetition::REQUIRED, Type::INT64);
+    node_ = PrimitiveNode::Make("column", Repetition::REQUIRED, TestType::type_num,
+        LogicalType::NONE, FLBA_LENGTH);
     schema_ = std::make_shared<ColumnDescriptor>(node_, 0, 0);
   }
 
   void SetUpSchemaOptional() {
-    node_ = PrimitiveNode::Make("int64", Repetition::OPTIONAL, Type::INT64);
+    node_ = PrimitiveNode::Make("column", Repetition::OPTIONAL, TestType::type_num,
+        LogicalType::NONE, FLBA_LENGTH);
     schema_ = std::make_shared<ColumnDescriptor>(node_, 1, 0);
   }
 
   void SetUpSchemaRepeated() {
-    node_ = PrimitiveNode::Make("int64", Repetition::REPEATED, Type::INT64);
+    node_ = PrimitiveNode::Make("column", Repetition::REPEATED, TestType::type_num,
+        LogicalType::NONE, FLBA_LENGTH);
     schema_ = std::make_shared<ColumnDescriptor>(node_, 1, 1);
   }
 
+  void GenerateData(int64_t num_values);
+
+  void SetupValuesOut();
+
   void SetUp() {
-    values_out_.resize(100);
+    SetupValuesOut();
     definition_levels_out_.resize(100);
```

Comment:
can you declare a constant for the size and use that everywhere 100 is currently hardcoded?